### ISSUE TITLE
[BUG] Error LNK2001 in Visual studio 2022

### ISSUE THREAD

**Issue by rty65tt** [google/benchmark#1879]
Today I decided to try benchmark in MS Visual Studio 2022 and I had problems, but I found a solution how fix it.

1. Open Solution Explorer > Right Click > Properties

3. Make sure that you have set dynamic linking:

Configuration Properties > C/C++ > Code Generation > Runtime Library > "Multi-threaded DLL (/MD)"" for Release and "Multi-threaded Debug DLL (/MDd)" for Debug

3. In Configuration Properties > C/C++ > Additional Include Directories - add path to \benchmark\include

4. Add in top you cpp file next code:
```
#define BENCHMARK_STATIC_DEFINE
#pragma comment(lib, "../../benchmark/build/src/Release/benchmark.lib")
#pragma comment (lib, "Shlwapi.lib" )
#include "../../benchmark/include/benchmark/benchmark.h"
```

**Comment by dmah42** [comment]
i wonder if there's something we need to bring into the library to avoid some of this.  the include is necessary, the library should be something you set in your project settings (iirc).  finally, BENCHMARK_STATIC_DEFINE will be set if BUILD_SHARED_LIBS is disabled in cmake, so you shouldn't need that either. :)

**Comment by rty65tt** [comment]
> i wonder if there's something we need to bring into the library to avoid some of this.
I don't know. :) Maybe a link to instructions on how to work with this benchmark in Visual Studio or Xcode without cmake.

I just have a C++ project in Visual Studio 2022 (without cmake). I wanted to measure part of the code with your benchmark.
I did as written in the instructions:

```
git clone https://github.com/google/benchmark.git
cd benchmark
cmake -E make_directory "build"
cmake -E chdir "build" cmake -DBENCHMARK_DOWNLOAD_DEPENDENCIES=on -DCMAKE_BUILD_TYPE=Release ../
cmake --build "build" --config Release
```

then added include and libraries to the project, but received ERROR LNK2001.

On stackoverflow I found a recommendation to add BENCHMARK_STATIC_DEFINE. It didn't help, I got the same error, because I had static linking set in my project settings. 
My spider sense told me to "change to dynamic" and it worked.
